core/scripts/local/BatchGasEstimation.js

```diff
@@ -0,0 +1,208 @@
+// This script Determines the max number of batched commits, reveals, and reward retrievals
+// that can fit into a single txn. This is done by creating a set of price requests and then
+// progressively committing, revealing and claiming rewards until the transaction reverts for
+// a given size. A binary search mechanism is used to reduce the number of iterations required
+// to find the max number that can fit within one transaction.
+
+const { RegistryRolesEnum } = require("../../../common/Enums.js");
+const { getRandomSignedInt, getRandomUnsignedInt } = require("../../../common/Random.js");
+const { moveToNextRound, moveToNextPhase } = require("../../utils/Voting.js");
+
+const Registry = artifacts.require("Registry");
+const Voting = artifacts.require("Voting");
+const VotingToken = artifacts.require("VotingToken");
+
+let results = {}; // Array to store maximum results achieved.
```

Comment:
Two nits:
1. Isn't this an object, not an array?
2. Slightly prefer not having global, mutable state if at all avoidable. It makes things harder to reason about IMO. Would prefer just declaring it in run and passing it to the methods or mutating with the return value.

Reply:
No problem. will refine.